### PR TITLE
Support new JAVIS lock, TS0601 roller blind motor. Fix controlling by position for TS0601_curtain

### PR DESCRIPTION
**Pull request by dinhchinh82**
This PR is related to : https://github.com/Koenkk/zigbee2mqtt/pull/4652

This PR support the following changes:
1.  Support new JAVIS lock
- Product : https://javishome.vn/khoa-thong-minh-js-slk2-zb-dung-cho-cua-go
- Model: JAVISLOCK
- Vendor: JAVIS
- Description Intelligent biometric digital lock
- Supports: battery
- open, close, operation (reporting only)
- Photo: https://javishome.vn/files/Khoa_Thong_Minh_2/khoa-thong-minh-js-slk2-zb-dung-cho-cua-go-2.jpg

2. Support new TS0601 roller blind motor:
 https://www.alibaba.com/product-detail/Zigbee-Roller-Shade-Switch-with-Motor_62231705367.html
This motor is similar to TS0601_curtain but the controlling by position is opposite so we could not use the same controller for both types.

3. Fix controlling by position for TS0601_curtain without adding invert_cover option.

### CODE REVIEW

File: converters/toZigbee.js [hunk 1]
```diff
@@ -2853,6 +2853,37 @@ const converters = {
...
+                    meta.logger.debug('_TZE200_wmcdj3aq: Curtain motor position is out of range');
+                }
+            } else if (key === 'state') {
+                value = value.toLowerCase();
+                switch (value) {
+                case 'close':
+                    await sendTuyaCommand(entity, 1025, 0, [1, 0]); // close
+                    break;
+                case 'open':
+                    await sendTuyaCommand(entity, 1025, 0, [1, 2]); // open
```
Comment (Koenkk) [inline converters/toZigbee.js:2875]: We can merge this with the existing converter, e.g. change `await sendTuyaCommand(entity, 1025, 0, [1, isTZE200_wmcdj3aq ? 1 : 0]); // 0x04 0x01: Open` and above define `const isTZE200_wmcdj3aq = entity.getDevice().manufacturerName === 'TZE200_wmcdj3aq`).

**Comment by Koenkk** [comment]
Note to self: after this merge https://github.com/Koenkk/zigbee2mqtt/pull/4652

File: converters/toZigbee.js (hunk 1 above)
Comment (dinhchinh82) [reply converters/toZigbee.js:2875]: @Koenkk we also considered about your solution at the beginning. By the way, there is a disadvantage of this solution. If there are some new similar models (manufatureName) like 'TZE200_wmcdj3aq' in the future, it is required to update the converter then the code of converter is mess. 

The root of this issue is TS0601 motor has some fingerPrint with opposite way to control the position. So my idea is separate them into 2 sub-categories. In the future, if there is any similar motor/roller bind like this, we just add new fingerPrint to the devices.js only.

Comment (zhouF96) [reply converters/toZigbee.js:2875]: Hi, @dinhchinh82 Would you please also move model ID "aabybja" and             
{modelID: 'TS0601', manufacturerName: '_TZE200_nkoabg8w'}, to use the new bind method? As they behaves the opposite from the current one.
Thanks very much!

Comment (dinhchinh82) [reply converters/toZigbee.js:2875]: @zhouF96 after @Koenkk merge this pull request, it's quite easy to add your fingerPrint device to the devices. js to have fully supported.

**Comment by Koenkk** [comment]
Found a way to merge both toZigbee converters. Let me know if it still works OK, then it can be merged.

File: converters/toZigbee.js (hunk 1 above)
Comment (zhouF96) [reply converters/toZigbee.js]: @dinhchinh82 Thanks very much, and will try it tomorrow morning.

**Comment by dinhchinh82** [comment]
@Koenkk It seems new updated code did not work.

For the TS0601_curtain, when I try to open the curtain from Home Assistant, there is an error from the log below:

```
Zigbee2MQTT:info  2020-10-15 00:47:45: Configuring '0xec1bbdfffe91e950'
Zigbee2MQTT:error 2020-10-15 00:47:51: Failed to call 'Receive' 'onZigbeeEvent' (TypeError: Cannot read property 'type' of undefined
    at /opt/zigbee2mqtt/lib/extension/receive.js:120:42
    at Array.filter (<anonymous>)
    at Receive.onZigbeeEvent (/opt/zigbee2mqtt/lib/extension/receive.js:119:65)
    at Controller.callExtensionMethod (/opt/zigbee2mqtt/lib/controller.js:372:44)
    at Controller.onZigbeeEvent (/opt/zigbee2mqtt/lib/controller.js:261:14)
    at Zigbee.emit (events.js:314:20)
    at Controller.<anonymous> (/opt/zigbee2mqtt/lib/zigbee.js:68:52)
    at Controller.emit (events.js:314:20)
    at Controller.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/controller.js:544:22)
    at Generator.next (<anonymous>))
```
For TS0601 blind motor, it is recognized as the light, not the cover as expected. Please check the screenshot from HA as below:
https://prnt.sc/uzc9pw

**Comment by Koenkk** [comment]
- Did you apply all updates files? fromZigbee/toZigbee and devices.js?
- It's discovered as a light because you defined it as such: https://github.com/Koenkk/zigbee2mqtt/pull/4652/files#diff-5cd90a075b18a077fab2f7f5a4066f63eade9d49e5ef0b023bc3878272786e9eR1248

**Comment by zhouF96** [comment]
Hi, Tested this morning. Still something wrong here. According to commit [8d64f1a](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/8d64f1a3688de1bd0e1b7352351df94a04fc1f73) the state of TS0601_roller_blind is still the opposite here include the operate about OPEN and CLOSE . And for OPEN command , I think it should be setting the position to 100%, which means expose the window here and the CLOSE should set the position to 0%, means let the curtain cover the window. (the behavior should sync to [lumi.curtain', 'lumi.curtain.aq2](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices.js#L1100)) currently all are the opposite for TS0601_roller_blind

**Comment by dinhchinh82** [comment]
> Hi, Tested this morning. Still something wrong here. According to commit [8d64f1a](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/8d64f1a3688de1bd0e1b7352351df94a04fc1f73) the state of TS0601_roller_blind is still the opposite here include the operate about OPEN and CLOSE . And for OPEN command , I think it should be setting the position to 100%, which means expose the window here and the CLOSE should set the position to 0%, means let the curtain cover the window. (the behavior should sync to [lumi.curtain', 'lumi.curtain.aq2](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices.js#L1100)) currently all are the opposite for TS0601_roller_blind

Could you please check the commit [4eef670](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/4eef67064401fad83f479362cb4066e972d45e3b) ? 
Your issue should be fixed in this commit.

**Comment by zhouF96** [comment]
> > Hi, Tested this morning. Still something wrong here. According to commit [8d64f1a](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/8d64f1a3688de1bd0e1b7352351df94a04fc1f73) the state of TS0601_roller_blind is still the opposite here include the operate about OPEN and CLOSE . And for OPEN command , I think it should be setting the position to 100%, which means expose the window here and the CLOSE should set the position to 0%, means let the curtain cover the window. (the behavior should sync to [lumi.curtain', 'lumi.curtain.aq2](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices.js#L1100)) currently all are the opposite for TS0601_roller_blind
> 
> @zhouF96 Could you please check the commit [4eef670](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/4eef67064401fad83f479362cb4066e972d45e3b) ?
> Your issue should be fixed in this commit.

to the device aabybja and {modelID: 'TS0601', manufacturerName: '_TZE200_nkoabg8w'} it seemed that using 
`                case 'close':
                    await sendTuyaCommand(entity, 1025, 0, [1,  2 ]); // close
                    break;
                case 'open':
                    await sendTuyaCommand(entity, 1025, 0, [1,  0 ]); // open
                    break;`
is right. What about the _TZE200_wmcdj3aq ?

**Comment by dinhchinh82** [comment]
@zhouF96 According to the declaration by fingerPrint, the '_TZE200_wmcdj3aq ' should work same as '_TZE200_nkoabg8w'
https://prnt.sc/uzlkrk

Please remove the old device from HA then pairing again to check.

**Comment by dinhchinh82** [comment]
> * Did you apply all updates files? fromZigbee/toZigbee and devices.js?
> * It's discovered as a light because you defined it as such: https://github.com/Koenkk/zigbee2mqtt/pull/4652/files#diff-5cd90a075b18a077fab2f7f5a4066f63eade9d49e5ef0b023bc3878272786e9eR1248

We also updated some code and tested again. Please merge the latest commit [4eef670](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/4eef67064401fad83f479362cb4066e972d45e3b).

Please note to merge some changes from Koenkk/zigbee2mqtt#4652

**Comment by zhouF96** [comment]
> @zhouF96 According to the declaration by fingerPrint, the '_TZE200_wmcdj3aq ' should work same as '_TZE200_nkoabg8w'
> https://prnt.sc/uzlkrk
> 
> Please remove the old device from HA then pairing again to check.

remove the old device from HA then pairing again to check and confirmed. the open and close is not the opposite to the original one ,but only the percentage of position. At least for '_TZE200_nkoabg8w'
for 'aabybja' I still need to test it.

**Comment by zhouF96** [comment]
> @zhouF96 According to the declaration by fingerPrint, the '_TZE200_wmcdj3aq ' should work same as '_TZE200_nkoabg8w'
> https://prnt.sc/uzlkrk
> 
> Please remove the old device from HA then pairing again to check.

Hi, also tested 'aabybja' same as '_TZE200_nkoabg8w' the original OPEN and CLOSE is right and the position [currently](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/4eef67064401fad83f479362cb4066e972d45e3b)is right. What about the '_TZE200_wmcdj3aq' ? I don't have this device. Thank you very much

**Comment by dinhchinh82** [comment]
> > @zhouF96 According to the declaration by fingerPrint, the '_TZE200_wmcdj3aq ' should work same as '_TZE200_nkoabg8w'
> > https://prnt.sc/uzlkrk
> > Please remove the old device from HA then pairing again to check.
> 
> Hi, also tested 'aabybja' same as '_TZE200_nkoabg8w' the original OPEN and CLOSE is right and the position [currently](https://github.com/Koenkk/zigbee-herdsman-converters/pull/1657/commits/4eef67064401fad83f479362cb4066e972d45e3b)is right. What about the '_TZE200_wmcdj3aq' ? I don't have this device. Thank you very much

_TZE200_wmcdj3aq works well but need to set the `reverse_direction` to be `true` from MQTT setting.

**Comment by Koenkk** [comment]
Is this ready for merge?

**Comment by dinhchinh82** [comment]
> Is this ready for merge?

Yes. Please :)

**Pull request merged**